### ISSUE TITLE
Problems with samples() function

### ISSUE THREAD

**Issue by BoPeng** [vatlab/VariantTools#118]
I have lately run different tests in Vtools and have tried to use the `sample()` function with no luck. 
We have tried to follow the instructions from the webpage `https://vatlab.github.io/vat-docs/documentation/functions/samples/`. 

Our genotype data has been imported with 
```
vtools import /vcf/variants.hFiltered.combined.vcf.gz --var_info DP --build hg19
```
and we have successfully performed different statistical tests amongst others SKAT. 

When I use the command:
```
vtools output variant chr pos ref alt -l 10
```
everything is ok, but when I add "samples()"
as this: 
```
vtools output variant chr pos ref alt "samples()" -l 10
```
the following ERROR message appears: 

```
ERROR: Failed to execute query. Field "samples()-l" might be misspecified
```

I have tried different versions of 
```
samples(), "samples()", samples ('sample_variant=aff=2'). 
```

Do you have any suggestions for solving these issues?

**Comment by BoPeng** [comment]
```
"samples()-l"
```
looks very strange as if  "sample()" absorbs the `-l` option. What do you see if you separate `samples()` and `-l 10`, like the following?

```
vtools output variant chr pos ref alt "samples()" alt -l 10
```

**Comment by grybeate** [comment]
When I try 
vtools output variant chr pos ref alt "samples()" alt -l 10 and 
vtools output variant chr pos ref alt "samples()" alt 
I get this ERROR: Failed to execute query. One or more fields might be misspecified: Failed to open genotype database

**Comment by BoPeng** [comment]
@jma7 Can you reproduce this?

**Comment by jma7** [comment]
I got 
```
ERROR: Failed to execute query. One or more fields might be misspecified: Failed to open genotype database
```
Are you sure "samples()" is a valid field to add into the command line for vtools output?
What are you trying to see? @BoPeng

**Comment by BoPeng** [comment]
This means this function tries to open the genotype database which is deprecated now.

**Comment by BoPeng** [comment]
@jma7 The `samples()` function is expanded to [samples("genotype.DB", parameters)](https://github.com/vatlab/varianttools/blob/master/src/variant_tools/utils.py#L3347), and is then handled by sqlite [here](https://github.com/vatlab/varianttools/blob/master/src/sqlite/vt_sqlite3_ext.cpp#L2290). In this function, vtools checks the genotype database, find the samples, and returns sample information. To make this work with the latest version, you will have to change the genotype DB to hdf DB, and calls the HDF5 C api to get the same information.

Do you know how to do this?

**Comment by BoPeng** [comment]
@grybeate We apologize but this function is no longer compatible with the recent versions of variant tools due to underlying changes of the data storage model. Obviously our test cases did not catch this and our documentation has not been updated. We are trying to fix it now but not sure how long it will take.

**Comment by jma7** [comment]
@BoPeng It will take some time to figure out the C API, is that the priority now? 
There are some other functions like "genotype()", we'd better do all of them together.

**Comment by BoPeng** [comment]
@jma7 Basically I only need you to write a C function outside of vtools that opens the hdf5 database and retrieve the information with given parameter. You can even ignore the additional query parameters for now.

**Comment by BoPeng** [comment]
@grybeate We re-implemented this feature and released variant tools 3.1.0. Please update to this version and let us know if it does not work for you.
